### PR TITLE
Dashboard tweak

### LINKED ISSUE #1647: Dashboard tweak
## Is your feature request related to a problem? Please describe 👀

I missed a couple of things when reviewing the dashboard change 
- The invoice widgets are now not just for invoices, and would be better if lifted out 
- The requisitions stats panel says 'Today' but does not filter by date, so it showing all new requisitions

While updating, we've had a further enhancement request. So, bundling all the changes together:

- [x] Change `New requisitions` to just be `Requisitions`
- [x] Change the `Today` under `New requisitions` to be `New`
- [x] Move `packages/invoices/src/OutboundShipment/Dashboard/Widget` to `packages/dashboard/src/widgets` and rename to `DistributionWidget`
- [x] Move `packages/invoices/src/InboundShipment/Dashboard/Widget` to `packages/dashboard/src/widgets` and rename to `ReplenishmentWidget`
- [x] Change the heading `Shipments that have not been shipped` to `Shipments`, AND
- [x] Change the wording `Today` under it to `Have not been shipped
- [x] Add `Internal orders` Stats panel with `Not sent` as a count
- [x] Add a badge on the internal orders menu item showing the count of orders which are not sent ( optional extra )

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1647 

# 👩🏻‍💻 What does this PR do? 
 

- [ ] Change `New requisitions` to just be `Requisitions`
- [ ] Change the `Today` under `New requisitions` to be `New`
- [ ] Move `packages/invoices/src/OutboundShipment/Dashboard/Widget` to `packages/dashboard/src/widgets` and rename to `DistributionWidget`
- [ ] Move `packages/invoices/src/InboundShipment/Dashboard/Widget` to `packages/dashboard/src/widgets` and rename to `ReplenishmentWidget`
- [ ] Change the heading `Shipments that have not been shipped` to `Shipments`, AND
- [ ] Change the wording `Today` under it to `Have not been shipped
- [ ] Add `Internal orders` Stats panel with `Not sent` as a count
- [ ] Add a badge on the internal orders menu item showing the count of orders which are not sent
 

image

# 🧪 How has/should this change been tested? 

Start site, check dashboard.

## 📃 Documentation

- [ ] Update screenshot of dashboard?

### CODE REVIEW

**Review by roxy-dao — CHANGES_REQUESTED** [review]
Most of the stats work great! Thanks Mark (: 

Just a few requests below 🙏🏻

File: client/packages/dashboard/src/api/hooks/statistics/useRequestCounts.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { useQuery } from '@openmsupply-client/common';
+import { useDashboardApi } from '../utils/useDashboardApi';
+
+export const useRequestCounts = () => {
+  const api = useDashboardApi();
+  return useQuery(api.keys.response(), api.get.requestRequisitionCounts, {
```
Comment (roxy-dao) [inline client/packages/dashboard/src/api/hooks/statistics/useRequestCounts.ts:6]:
```suggestion
  return useQuery(api.keys.request(), api.get.requestRequisitionCounts, {
```
(:

File: client/packages/dashboard/src/api/hooks/utils/useDashboardApi.ts [hunk 2]
```diff
@@ -8,6 +8,9 @@ export const useDashboardApi = () => {
     count: () => [...keys.base(), 'count', storeId] as const,
     items: () => [...keys.count(), 'items'] as const,
     stock: () => [...keys.count(), 'stock'] as const,
+    response: () => [...keys.count(), 'response'] as const,
+    outbound: () => [...keys.count(), 'outbound'] as const,
+    inbound: () => [...keys.count(), 'inbound'] as const,
```
Comment (roxy-dao) [inline client/packages/dashboard/src/api/hooks/utils/useDashboardApi.ts:13]: Missing request (:

File: client/packages/dashboard/src/widgets/ReplenishmentWidget.tsx [hunk 3]
```diff
@@ -11,15 +11,22 @@ import {
...
-  const t = useTranslation(['app', 'dashboard']);
+  const t = useTranslation('dashboard');
   const formatNumber = useFormatNumber();
-  const { data, isLoading, isError, error } = useInbound.utils.counts();
+  const { data, isLoading, isError, error } = useDashboard.statistics.inbound();
+  const {
+    data: requestCount,
+    isLoading: isRequestCountLoading,
+    isError: isRequestCountError,
+    error: responseCountError,
```
Comment (roxy-dao) [inline client/packages/dashboard/src/widgets/ReplenishmentWidget.tsx:28]:
```suggestion
    error: requestCountError,
```
(:

File: client/packages/host/src/components/Navigation/ReplenishmentNav.tsx [hunk 4]
```diff
@@ -42,6 +45,7 @@ export const ReplenishmentNav: FC = () => {
               .addPart(AppRoute.InternalOrder)
               .build()}
             text={t('internal-order')}
+            badgeProps={getBadgeProps(requestCount?.draftCount)}
```
Comment (roxy-dao) [inline client/packages/host/src/components/Navigation/ReplenishmentNav.tsx:48]: The badge looks great. Find it a bit weird to have it just for `Internal Orders` though... 
https://github.com/openmsupply/open-msupply/issues/1647#issuecomment-1529296336

File: server/service/src/dashboard/requisition_count.rs [hunk 5]
```diff
@@ -13,6 +13,14 @@ pub trait RequisitionCountServiceTrait: Send + Sync {
     ) -> Result<i64, RepositoryError> {
         RequisitionCountService {}.new_response_requisition_count(ctx, store_id)
     }
+
+    fn draft_request_requisition_count(
```
Comment (roxy-dao) [inline server/service/src/dashboard/requisition_count.rs:17]: Could you refactor these so requisition_count returns both response and request counts in api? 
`invoice_counts.rs` has an example of this with outbound/inbound counts~

**Comment by mark-prins** [comment]
Thanks @roxy-dao - have made the changes. I've gone with a single requisition count query, so that we only need to make the single api call from the f/e. Ideally the response would have separate permissions for each part (request / response) : since they are both using the `Resource::RequisitionStats` permission, I have that at the top level. the query returns an error if you don't have that permission, and both widgets fail to show.

**Review by roxy-dao — APPROVED** [review]
LGTM~ Thanks Mark (: 

Just some conflicts to deal with~

**Pull request merged**